src/lib/y2firstboot/clients/wsl_product_selection.rb

```diff
@@ -0,0 +1,147 @@
+    class WSLProductSelection < Yast::Client
+      # Runs the client
+      #
+      # @throw [RuntimeError] see {#require_registration}.
+      #
+      # @return [Symbol]
+      def run
+        require_registration
+
+        return :next if products.none?
+
+        dialog = Dialogs::WSLProductSelection.new(products,
+          default_product: product,
+          wsl_gui_pattern: wsl_gui_pattern?)
+
+        result = dialog.run
+
+        save(product: dialog.product, wsl_gui_pattern: dialog.wsl_gui_pattern) if result == :next
+
+        result
+      end
+
+    private
+
+      WSL_GUI_PATTERN = "wsl_gui".freeze
+      private_constant :WSL_GUI_PATTERN
+
+      # Saves changes
+      #
+      # @param product [Hash] Selected product
+      # @param wsl_gui_pattern [Boolean] Whether to install WSL GUI pattern
+      def save(product:, wsl_gui_pattern:)
+        self.product = product
+        self.wsl_gui_pattern = wsl_gui_pattern
+        update_registration
+      end
+
+      # Product to use
+      #
+      # @see {ẂSLConfig}
+      #
+      # @return [Hash]
+      def product
+        WSLConfig.instance.product || default_product
+      end
+
+      # Sets the product to use
+      #
+      # @see {ẂSLConfig}
+      #
+      # @param value [Hash] A product
+      def product=(value)
+        WSLConfig.instance.product = value
+      end
+
+      # Whether the WSL GUI pattern should be installed
+      #
+      # @see {ẂSLConfig}
+      #
+      # @return [Boolean]
+      def wsl_gui_pattern?
+        WSLConfig.instance.patterns.include?(WSL_GUI_PATTERN)
+      end
+
+      # Sets whether to install the WSL GUI pattern
+      #
+      # @param value [Boolean]
+      def wsl_gui_pattern=(value)
+        if value
+          WSLConfig.instance.patterns.push(WSL_GUI_PATTERN).uniq!
+        else
+          WSLConfig.instance.patterns.delete(WSL_GUI_PATTERN)
+        end
+      end
+
+      # Updates values stored in registration
+      #
+      # Those values indicates to registration what product was selected and whether the product
+      # has to be registered.
+      #
+      # @see {Registration::Storage::InstallationOptions}
+      def update_registration
+        yaml_product = WSLConfig.instance.product
+        force_registration = WSLConfig.instance.product_switched? || wsl_gui_pattern?
+
+        Registration::Storage::InstallationOptions.instance.yaml_product = yaml_product
+        Registration::Storage::InstallationOptions.instance.force_registration = force_registration
+      end
+
+      # Name of the default product to use from YAML file
+      #
+      # @return [String]
+      def default_product
+        return nil if products.none?
+
+        products.find { |p| p["default"] } || products.first
+      end
+
+      # All products from YAML file
+      #
+      # @return [Array<Hash>]
+      def products
+        @products ||= Registration::YamlProductsReader.new("/tmp/products.yml").read
```

Comment:
That looks like a forgotten testing path?

Reply:
Arf